c++/src/kj/compat/url.h

```diff
@@ -124,10 +133,13 @@ struct Url {
     // path, and query, but omits userInfo (which should be used to construct the Authorization
     // header) and fragment (which should not be transmitted).
 
-    HTTP_REQUEST
+    HTTP_REQUEST,
     // The path to place in the first line of a regular HTTP request. This includes only the path
     // and query. Scheme, user, host, and fragment are omitted.
 
+    ARBITRARY,
+    // An arbitrary URL that may omit any component.
```

Comment:
So, what does this mean exactly? Is it meant to parse a relative URL? Or is it really an _absolute_ URL that can omit any component?

If it's an absolute URL missing components, can you name any existing applications where such a thing today? I feel like this isn't a well-defined concept and I'm not really sure if we should be inventing it from scratch.

If it's meant to be a relative URL, then I'm not sure if this is the right approach. Generally, the right way to parse a relative URL is by first obtaining the base URL, and then calling `parseRelative()` on that. I don't think the URL parser is set up to parse a relative URL without a base to evaluate it against. Two specific problems that I think come up here:

* How is a leading `../` handled? When parsing a URL relative to an existing URL, we evaluate these, so the final result doesn't contain `..`. But when parsing a truly relative URL I think you'd want to preserve them for later evaluation?
* Let's say we want to extend this class in the future to support non-HTTP URL schemes. Different schemes might have completely different parsing rules, right? So how can we parse a relative URL without as scheme, unless we have a base URL to evaluate against which defines the scheme?

Basically, I'm uncomfortable as I don't think this class was ever intended to represent a relative URL and it's not obvious to me that extending it that way won't lead to problems.

Sorry to bring this up late in the process. In the future we should probably try to discuss major design changes like this before code is written.